lib/rack/param_to_cookie.rb

```diff
@@ -33,6 +34,9 @@ def call env
         # check whether there's a new value for the cookie with this request
         params_value = req.params[param] rescue nil
 
+        # validate the length of the value
+        next if req.params[param].length > options[:max_length] rescue nil
```

Comment:
Is there a reason not to use `params_value` here? It looks like we've already handled potential errors above, so we should now be fairly confident that we have either a string or `nil`.

Also, as written this will skip the rest of the function if the request parameter is invalid. I don't see a reason that we couldn't continue with `params_value` `nil` if it was invalid, and still set the value for the cookie?

So... could this line just be
```
params_value = nil if params_value && params_value.length > options[:max_length]
```
?